src/common/utils/dateFormatter.ts

```diff
@@ -0,0 +1,41 @@
+export const isBefore = (date: Date | string, dateToCompare: Date | string): boolean => {
+  if (typeof date === 'string') date = new Date(date);
+  if (typeof dateToCompare === 'string') dateToCompare = new Date(dateToCompare);
+  return date.getTime() < dateToCompare.getTime();
+};
+
+export const isAfter = (date: Date | string, dateToCompare: Date | string): boolean => {
+  if (typeof date === 'string') date = new Date(date);
+  if (typeof dateToCompare === 'string') dateToCompare = new Date(dateToCompare);
+  return date.getTime() > dateToCompare.getTime();
+};
+
+export const differenceInSeconds = (laterDate: Date | string, earlierDate: Date | string): number => {
+  if (typeof laterDate === 'string') laterDate = new Date(laterDate);
+  if (typeof earlierDate === 'string') earlierDate = new Date(earlierDate);
+  return Math.floor((laterDate.getTime() - earlierDate.getTime()) / 1000);
+};
+
+export const subMinutes = (date: Date | string, amount: number): Date => {
+  if (typeof date === 'string') date = new Date(date);
+  const newDate = new Date();
+  newDate.setTime(date.getTime() - amount * 60 * 1000);
+  return newDate;
+};
+
+export const format = (date: Date | string, formatStr: string): string => {
+  if (typeof date === 'string') date = new Date(date);
+  const days = ['월', '화', '수', '목', '금', '토', '일'];
+  const formatter: { [key: string]: string } = {
+    M: (date.getMonth() + 1).toString(),
+    dd: date.getDate().toString().padStart(2, '0'),
+    E: days[date.getDay() - 1],
+    EEEE: days[date.getDay() - 1] + '요일',
```

Comment:
이렇게 하니 일요일이 undefined 라고 뜨네요
getDay는 0~6을 반환하고 0이 곧 일요일인데 -1을 해주니 -1이 되어서 undefined가 뜨게 돼요

<img width="308" alt="스크린샷 2024-10-17 오후 4 37 52" src="https://github.com/user-attachments/assets/8054a8b1-936b-4860-b2c4-74737f8dfed8">


```suggestion
  const days = ['일', '월', '화', '수', '목', '금', '토'];
  const formatter: { [key: string]: string } = {
    M: (date.getMonth() + 1).toString(),
    dd: date.getDate().toString().padStart(2, '0'),
    E: days[date.getDay()],
    EEEE: days[date.getDay()] + '요일',
```

위와 같이 수정하면 될 거 같습니다 :)

Reply:
🐞잡아주셔서 감사합니다!